### PR TITLE
Replaced jsonfield2 with jsonfield

### PR DESCRIPTION
**Pull request by iamsobanjaved**
**TL;DR -** Replaced jsonfield2 with jsonfield

JIRA: [BOM-2414](https://openedx.atlassian.net/browse/BOM-2414)

**What changed?**

We are going to replace jsonfield2 with jsonfield as the earlier one has archived and merged back in the latter one.

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

**Comment by iamsobanjaved** [comment]
We had to merge all jsonfield PRs in edX packages at once because this PR will bring latest version of jsonfield in platform as well which needs migration in few other packages as well.

**Comment by jmbowman** [comment]
@edx/masters-devs-gta Could someone from the owning squad please review this?  It's part of a coordinated model field package upgrade across several edx-platform dependencies.

**Review by jnlapierre — APPROVED** [review]
LGTM
